vignettes/post-hoc_differential-expression_bootstrap_microarray.Rmd

```diff
@@ -0,0 +1,269 @@
+
+```{r contrast-1}
+tested_variable <- "fev1_fvc_utah"
+C <- matrix(1*(colnames(X) == tested_variable), nrow = 1)
+rownames(C) <- "association_lung_func"
+colnames(C) <- colnames(X)
+C
+```
+
+For each gene $j$, we fit a model to estimate a vector of parameters $\beta_j$. Then we test for each gene $j$ and for the contrast $c$, if $\mathcal{H}_0: c\beta_j = 0$. In our case, we test if the parameter of the studied variable `fev1_fvc_utah` is equal to 0. 
+We want to obtain post hoc guarantees on a set of genes which are associated with lung function (contrast `fev1_fvc_utah`). We first build a `SansSouci` class object given the gene expression matrix $Y$, the design matrix $X$ and the contrast matrix $C$. Then we fit the model and calibrate the post hoc bound using the method `fit()`, setting the target risk to $\alpha = 0.1$. In the code below, we use only $B = 100$ bootstraps to save computing time. In practice, we recommend taking $B = 1000$ bootstraps for more stable results. 
+
+
+```{r fit-calibration-1}
+alpha = 0.1
+obj <- SansSouci(Y = Y, X = X, Contrast = C)
+res <- fit(obj, alpha = alpha, B = 100) # B=100 to save computing time, but should use B=1000.
+```
+
+For comparison purposes we also run the (parametric) Simes method introduced by Goeman and Solari (2011):
+
+
+```{r fit-Simes-1}
+res_Simes <- fit(obj, B = 0, family = "Simes", alpha = alpha) ## B=0 => no calibration!
+resList <- list("SansSouci" = res,
+                "Simes (parametric)" = res_Simes)
+```
+
+
+
+## Confidence curve
+
+In the absence of prior information on genes, a natural idea is to rank them by decreasing statistical significance, and a natural question to ask is:
+
+> Can we provide a lower confidence curve on the number (or proportion) of truly differentially expressed genes among the most significant genes?
+
+The confidence curve obtained by calibration is the solid black line in the figure below:
+```{r conf-curve-1}
+conf_bounds <- lapply(resList, predict, all = TRUE)
+cols <- c("black", "darkgray")
+p <- plotConfCurve(conf_bounds, xmax = 5000, cols = cols) + 
+  geom_line(linewidth = 1.5)
+p
+```
+
+This plot can be interpreted as follows: among the 2000 most significant genes, the number of truly differentially expressed genes is at least 1060 (right panel). Equivalently, the FDP among these 2000 genes is at most 0.47 (left panel).
+
+The dashed gray curve is obtained by the parametric Simes method introduced by Goeman and Solari (2011). The comparison between the two curves illustrates the gain in power obtained by using permutation methods to adapt to the dependence between genes. In this example, the parametric Simes method gives the following guarantees: among the 2000 most significant genes, the number of truly differentially expressed genes is at least 492 (right panel). Equivalently, the FDP among these 2000 genes is at most 0.754 (left panel).
+
```

Comment:
Here can you avoid hard coded figures ie replace all numbers by the result of  R expressions?
This is necessary for internal  consistency and reproducibility